monitoring/src/monitoring_service/handlers.py

```diff
@@ -0,0 +1,300 @@
+from dataclasses import dataclass  # isort:skip noqa differences between python 3.6 and 3.7
+from typing import List, cast
+
+import structlog
+from web3 import Web3
+
+from monitoring_service.database import Database
+from monitoring_service.events import (
+    ActionClaimRewardTriggeredEvent,
+    ActionMonitoringTriggeredEvent,
+    Event,
+    ReceiveChannelClosedEvent,
+    ReceiveChannelOpenedEvent,
+    ReceiveChannelSettledEvent,
+    ReceiveNonClosingBalanceProofUpdatedEvent,
+    ReceiveTokenNetworkCreatedEvent,
+    ScheduledEvent,
+    UpdatedHeadBlockEvent,
+)
+from monitoring_service.states import Channel, MonitoringServiceState
+from raiden_contracts.constants import CONTRACT_MONITORING_SERVICE, ChannelState
+from raiden_contracts.contract_manager import ContractManager
+
+log = structlog.get_logger(__name__)
+
+
+@dataclass
+class Context:
+    ms_state: MonitoringServiceState
+    db: Database
+    scheduled_events: List[ScheduledEvent]
+    w3: Web3
+    contract_manager: ContractManager
+    last_known_block: int
+
+
+class EventHandler:
+    """ Base class for event handlers.
+
+    An event handler needs to be idempotent as events can be triggered
+    multiple times.
+    """
+    def handle_event(self, event: Event):
+        raise NotImplementedError
+
+
+@dataclass
+class TokenNetworkCreatedEventHandler(EventHandler):
+    context: Context
+
+    def handle_event(self, event: Event):
+        if isinstance(event, ReceiveTokenNetworkCreatedEvent):
```

Comment:
Does mypy understand
```python
    assert isinstance(event, ReceiveTokenNetworkCreatedEvent)
```
? That would give us an error in the bad case and reduce the nesting level.

Reply:
Yes, it does. Changed as well.